src/stactools/core/utils/raster_footprint.py

```diff
@@ -208,23 +225,32 @@ def data_footprint(
     if no_data is None:
         no_data = src.nodata
 
-    arr = src.read(1, out_shape=src.shape)
-    data_val = 1 if no_data != 1 else 0
+    if not src.indexes:
+        raise ValueError(
+            "Raster footprint cannot be computed for an asset with no bands."
+        )
+    if not bands:
+        bands = src.indexes
+
+    data: npt.NDArray[np.uint8] = np.full(src.shape, 0, dtype=np.uint8)
+    for index in bands:
+        band_data = src.read(index, out_shape=src.shape)
 
-    if no_data is not None:
-        if numpy.isnan(no_data):
-            arr[~numpy.isnan(arr)] = data_val
+        if no_data is not None:
+            if np.isnan(no_data):
+                data[~np.isnan(band_data)] = 1
+            else:
+                data[np.where(band_data != no_data)] = 1
         else:
-            arr[numpy.where(arr != no_data)] = data_val
-    else:
-        arr.fill(data_val)
+            data.fill(1)
+            break
```

Comment:
Why the `break` here?

Reply:
If all the pixels are `1`, then the data array can't be modified in any future loop (we're OR-ing the pixels, not AND-ing them).

Now that I look at it though, we could just skip the loop all together since this branch only hits if there is no nodata value, it doesn't care about the band data. I'll fix.